### DISCUSSION TITLE
Container Tree Refinement Template in V4

### DISCUSSION THREAD

**Discussion by soumyadeep2207** [microsoft-search/pnp-modern-search#2892] [Q&A]
Hi All,

Compare to V3 in V4 there is very less number of refinement template available. Our customer was using Container Tree template to browse through the folder structure ( nested folder structure) as there exist items in pretty deep in the folder levels. Any suggestion or alternative to have that in V4.

Also can we create our own refinement template and associate with the existing solution

Looking forward to your inputs

Thanks,
Soumyadeep

**Comment by kasperbolarsen** [comment]
As far as I know there are two options: 
Check if the commercial solution from Sword Group contains the option (https://www.aequos.ca/) or build it yourself using the options available in the project (https://microsoft-search.github.io/pnp-modern-search/extensibility/)

**Reply by soumyadeep2207** [reply]
Can we create Refinement Template in Search Filter Webpart using Extensibility project? This is from documentation "Custom layouts are only supported for the 'Search Results' Web Part. You can't add custom layout for the 'Search Filters' Web Part."

**Reply by soumyadeep2207** [reply]
![image](https://user-images.githubusercontent.com/55022369/232184310-e0be227e-844c-4005-a38f-4059be051af1.png)
All these refinerment template  were there in V3 but in V4 there are very less number of refinement template. Can we create those in V4  or can we have those in V4?

Any reply will be helpful

**Reply by wobba** [reply]
There has been requests for tree view filter templated, but we need someone to develop and contribute it to the project.
